android/tangram/src/com/mapzen/tangram/MapData.java

```diff
@@ -1,161 +1,154 @@
-/* ----------------------------------------------------------------------------
- * This file was automatically generated by SWIG (http://www.swig.org).
- * Version 3.0.7
- *
- * Do not make changes to this file unless you know what you are doing--modify
- * the SWIG interface file instead.
- * ----------------------------------------------------------------------------- */
-
 package com.mapzen.tangram;
 
+import com.mapzen.tangram.geometry.Geometry;
+import com.mapzen.tangram.geometry.Point;
+import com.mapzen.tangram.geometry.Polygon;
+import com.mapzen.tangram.geometry.Polyline;
+
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Map;
 
-public class MapData extends DataSource {
-    private transient long swigCPtr;
-    private boolean swigCMemOwnDerived;
+/**
+ * {@code MapData} is a named collection of drawable map features.
+ */
+public class MapData {
 
-    protected MapData(long cPtr, boolean cMemoryOwn) {
-        super(TangramJNI.MapData_SWIGSmartPtrUpcast(cPtr), true);
-        swigCMemOwnDerived = cMemoryOwn;
-        swigCPtr = cPtr;
-    }
+    String name;
+    long pointer = 0;
+    MapController map;
+    List<String> geojson = new ArrayList<>();
+    List<Geometry> features = new ArrayList<>();
 
-    protected static long getCPtr(MapData obj) {
-        return (obj == null) ? 0 : obj.swigCPtr;
+    /**
+     * Construct a collection of drawable map features.
+     * @param name The name of the data collection. Once added to a map, features from this
+     * {@code MapData} will be available from a data source with this name, just like a data source
+     * specified in a scene file.
+     */
+    public MapData(String name) {
+        this.name = name;
     }
 
-    protected void finalize() {
-        delete();
+    /**
+     * Get the name of this {@code MapData}.
+     * @return The name.
+     */
+    public String name() {
+        return name;
     }
 
-    public synchronized void delete() {
-        if (swigCPtr != 0) {
-            if (swigCMemOwnDerived) {
-                swigCMemOwnDerived = false;
-                TangramJNI.delete_MapData(swigCPtr);
-            }
-            swigCPtr = 0;
+    /**
+     * Add the features from this {@code MapData} to a map.
+     * <p>
+     * This {@code MapData} will be associated with the given map until {@link #removeFromMap()}
+     * is called or until this method is called again.
+     * @param map The {@code MapController} managing the destination map.
+     */
+    public void addToMap(MapController map) {
+        if (map == null) {
+            throw new RuntimeException("Cannot assign a null map, use removeFromMap() instead");
```

Comment:
Also I do understand the association of `MapController` and `MapData` here, but I think it will be more intuitive for `MapController` to own a set of `MapData(s)` references? Your thoughts?